x-pack/plugins/alerts/server/health/get_state.test.ts

```diff
@@ -5,73 +5,187 @@
 
-describe('getHealthStatusStream()', () => {
-  const mockTaskManager = taskManagerMock.createStart();
-
-  it('should return an object with the "unavailable" level and proper summary of "Alerting framework is unhealthy"', async () => {
-    mockTaskManager.get.mockReturnValue(
-      new Promise((_resolve, _reject) => {
-        return {
-          id: 'test',
-          attempts: 0,
-          status: TaskStatus.Running,
-          version: '123',
-          runAt: new Date(),
-          scheduledAt: new Date(),
-          startedAt: new Date(),
-          retryAt: new Date(Date.now() + 5 * 60 * 1000),
-          state: {
-            runs: 1,
-            health_status: HealthStatus.Warning,
-          },
-          taskType: 'alerting:alerting_health_check',
-          params: {
-            alertId: '1',
-          },
-          ownerId: null,
-        };
+const tick = () => new Promise((resolve) => setImmediate(resolve));
+
+const getHealthCheckTask = (overrides = {}): ConcreteTaskInstance => ({
+  id: 'test',
+  attempts: 0,
+  status: TaskStatus.Running,
+  version: '123',
+  runAt: new Date(),
+  scheduledAt: new Date(),
+  startedAt: new Date(),
+  retryAt: new Date(Date.now() + 5 * 60 * 1000),
+  state: {
+    runs: 1,
+    health_status: HealthStatus.OK,
+  },
+  taskType: 'alerting:alerting_health_check',
+  params: {
+    alertId: '1',
+  },
+  ownerId: null,
+  ...overrides,
+});
+
+describe('getHealthServiceStatusWithRetryAndErrorHandling', () => {
+  beforeEach(() => jest.useFakeTimers());
+
+  it('should get status at each interval', async () => {
+    const mockTaskManager = taskManagerMock.createStart();
+    mockTaskManager.get.mockResolvedValue(getHealthCheckTask());
+    const pollInterval = 100;
+    const halfInterval = Math.floor(pollInterval / 2);
+
+    interval(pollInterval)
+      .pipe(switchMap(() => getHealthServiceStatusWithRetryAndErrorHandling(mockTaskManager)))
+      .subscribe();
+
+    // shouldn't fire before poll interval passes
+    // should fire once each poll interval
+    jest.advanceTimersByTime(halfInterval);
+    expect(mockTaskManager.get).toHaveBeenCalledTimes(0);
+    jest.advanceTimersByTime(halfInterval);
+    expect(mockTaskManager.get).toHaveBeenCalledTimes(1);
+    jest.advanceTimersByTime(pollInterval);
+    expect(mockTaskManager.get).toHaveBeenCalledTimes(2);
+    jest.advanceTimersByTime(pollInterval);
+    expect(mockTaskManager.get).toHaveBeenCalledTimes(3);
+  });
+
+  it('should retry on error', async () => {
+    const mockTaskManager = taskManagerMock.createStart();
+    mockTaskManager.get.mockRejectedValue(new Error('Failure'));
+    const retryDelay = 10;
+    const pollInterval = 100;
+    const halfInterval = Math.floor(pollInterval / 2);
+
+    interval(pollInterval)
+      .pipe(
+        switchMap(() =>
+          getHealthServiceStatusWithRetryAndErrorHandling(mockTaskManager, retryDelay)
+        )
+      )
+      .subscribe();
```

Comment:
Is it worth using `getHealthStatusStream` directly here with the interval being an argument (we can use default value in the implementation)?

That way the unit tests ensure the composition is behaving as expected.
Just incase someone changes the `switchMap` in `getHealthStatusStream` in the future to something that behaved differently... 🤔